kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/queues/MeteredSchedulingQueue.java

```diff
@@ -0,0 +1,66 @@
+package dev.responsive.kafka.api.async.internals.queues;
+
+import dev.responsive.kafka.api.async.internals.events.AsyncEvent;
+import dev.responsive.kafka.api.async.internals.metrics.AsyncProcessorMetricsRecorder;
+
+public class MeteredSchedulingQueue<KIn> implements SchedulingQueue<KIn> {
+  private final AsyncProcessorMetricsRecorder metricsRecorder;
+  private final SchedulingQueue<KIn> wrapped;
+
+  public MeteredSchedulingQueue(
```

Comment:
I wound up measuring the scheduling metrics by wrapping it in this class and calling `recordX` methods. Alternatively, I could have set up gauges that sample the sizes directly. I chose this approach so I wouldn't have to make `SchedulingQueue` thread-safe